core/database/foxx/api/repository/validation.js

```diff
@@ -0,0 +1,226 @@
+"use strict";
+
+const { Result } = require("./types");
+const g_lib = require("../support");
+
+/**
+ * Standalone validation functions following Rust patterns
+ * Pure functions that return Result types for error handling
+ *
+ * See: https://doc.rust-lang.org/book/ch03-03-how-functions-work.html
+ * Functions in Rust are expressions that can return values
+ *
+ * See: https://doc.rust-lang.org/book/ch09-00-error-handling.html
+ * Rust emphasizes explicit error handling through Result types
+ */
+
+// Validate that a value is a non-empty string
+// Reusable helper following DRY principle
+const validateNonEmptyString = (value, fieldName) => {
+    if (!value || typeof value !== "string" || value.trim() === "") {
+        return Result.err({
+            code: g_lib.ERR_INVALID_PARAM,
+            message: `${fieldName} is required and must be a non-empty string`,
+        });
+    }
+    return Result.ok(true);
+};
+
+// Validate common repository fields
+// Pure function - no side effects, deterministic output
+const validateCommonFields = (config) => {
```

Comment:
This function is very cleanly implemented, I like how the validateNonEmptyString is reusued and the error messages are stacked before they are thrown.